pkg/proxy/proxy.go

```diff
@@ -51,6 +53,14 @@ func (p *SprayProxy) HandleProxy(c *gin.Context) {
 		zap.String("query", c.Request.URL.RawQuery),
 		zap.Bool("insecure-tls", p.insecureTLS),
 		zap.String("request-id", c.GetString("requestId")),
+		zap.Bool("insecure-webhook", p.insecureWebhook),
+	}
+	// validate incoming request
+	if err := validateProxyRequest(c.Request, p.insecureWebhook); err != nil {
+		// we do not want to expose internal information, so returning generic failure message
+		c.String(http.StatusForbidden, "forbidden: request validation failure")
```

Comment:
I think returning a `400 Bad Request` code would be sufficient.

Reply:
Updated.